### PR TITLE
DT-1773 All validation done by validator

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ services:
     image: 'bitnami/redis:5.0'
     networks:
       - hmpps
-    container_name: redis 
+    container_name: redis
     environment:
       - ALLOW_EMPTY_PASSWORD=yes
```
Comment (andymarke) [inline docker-compose.yml:10]: dunno what has happened here

File: server/routes/courtRegister/addNewCourtDetailsValidator.test.ts [hunk 2]
```diff
@@ -44,13 +44,17 @@ describe('addNewCourtDetailsValidator', () => {
       const form = { ...validForm, id: 'A' }
       const nextPage = await validate(form, req, lookup)
       expect(nextPage).toEqual('/court-register/add-new-court-details')
-      expect(req.flash).toBeCalledWith('errors', [{ href: '#id', text: 'Court code must be at least 2 characters' }])
+      expect(req.flash).toBeCalledWith('errors', [
+        { href: '#id', text: 'Enter a court code between 2 and 12 characters' },
+      ])
```
Comment (andymarke) [inline server/routes/courtRegister/addNewCourtDetailsValidator.test.ts:49]: decided that GDS advice about reducing error message variations (which works well with the validator)

File: server/validation/validation.ts [hunk 3]
```diff
@@ -13,14 +13,15 @@ Validator.register(
   },
   'Enter a real postcode, like AA11AA'
 )
+type lookupFn<T> = (code: string) => Promise<T | null>
```
Comment (andymarke) [inline server/validation/validation.ts:16]: added for readability

File: server/validation/validation.ts [hunk 4]
```diff
@@ -30,7 +31,7 @@ export function validateAsync<T>(
     'unique-subcode',
     async (value, allowedBuildingId, request, passes) => {
       if (typeof value === 'string') {
-        if (!isBlank(value)) {
+        if (isCourtOrBuildingCode(value)) {
```
Comment (andymarke) [inline server/validation/validation.ts:34]: we can only validate codes that pass the lookups own validation, hence check size before doing additional check.
Need to find a better way to  short circuit this validation without this additional check

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**